### PR TITLE
[DUOS-344][risk=no] Standardize the Users table.

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-344

* Migrate`dacuser` table to `users`; `user` is a reserved word in postgresql so we can't use that.
* Migrate `dacuserid` to `user_id`
* Migrate `displayname` to `display_name`
* Migrate `createdate` to `create_date`
* Significant DAO refactoring. There are a number of cases where other tables have a `user_id` column so affected queries needed some rework.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by JVThomas** [comment]
I noticed this issue when pairing the API with the UI. The `currentUser` data in Storage is carrying a user object with `userId` present and not `dacUserId`, causing validation/rendering issues wherever `dacUserId` is expected.

In the example below, the vote button never shows up on the Chair Console because it can't check `currentUser.dacUserId` against `vote.dacUserId` (essentially undefined === vote.dacUserId check).

Screen Shot 2022-06-27 at 11 54 34 AM

**Comment by rushtong** [comment]
> I noticed this issue when pairing the API with the UI. The `currentUser` data in Storage is carrying a user object with `userId` present and not `dacUserId`, causing validation/rendering issues wherever `dacUserId` is expected.

Oof, yeah, I missed some legacy population when doing some cleanup. These changes ensure that we always set the value of `dacUserId` which fixes the underlying problem

File: src/main/java/org/broadinstitute/consent/http/service/UserService.java [hunk 1]
```diff
@@ -63,19 +63,27 @@ public UserService(UserDAO userDAO, UserPropertyDAO userPropertyDAO, UserRoleDAO
...
     public static class SimplifiedUser {
+        public Integer userId;
+        @Deprecated
         public Integer dacUserId;
         public String displayName;
         public String email;
 
         public SimplifiedUser(User user) {
+            this.userId = user.getUserId();
+            this.dacUserId = user.getUserId();
```
Comment (quazi-h) [inline src/main/java/org/broadinstitute/consent/http/service/UserService.java:74]: Was wondering if retaining the dacUserId here was intentional (lines 68, 74, 87, 88)

**Review by quazi-h — COMMENTED** [review]
I found a few files where "dacUserId" showed up when searching the consent files.

yaml files:

api-docs.yaml
line #3402 dacuserId -> userId
line #3513 dacUserId -> userId

User.yaml
 line #6 dacUserId -> userId
 
Vote.yaml
 line #10 dacUserId -> userId
 
 Working my way through the test and other project files still.

File: src/main/java/org/broadinstitute/consent/http/service/UserService.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/UserService.java:74]: This is very intentional. Until the UI can rely on only `userId`, we need to continue populating the `dacUserId` value.

**Comment by rushtong** [comment]
> I found a few files where "dacUserId" showed up when searching the consent files.
> ...
> Vote.yaml line #10 dacUserId -> userId
> 
> Working my way through the test and other project files still.

There are two objects, `Vote` and `Match` that still have a `dacUserId` column. This PR doesn't try to address those, just the usages of User.dacUserId

**Comment by rushtong** [comment]
> I found a few files where "dacUserId" showed up when searching the consent files.
> 
> yaml files:
> 
> api-docs.yaml line #3402 dacuserId -> userId line #3513 dacUserId -> userId
> 
> User.yaml line #6 dacUserId -> userId
> 

Nice catch - I'll update the User.yaml and double check api-docs.yaml

**Comment by rushtong** [comment]
@quazi-broad - All existing references to `dacUserId` or `dacuserId` in the api-docs are intentional. In the User case, we need to continue populating that field until the UI is migrated over. In the other cases, there are other objects that have not been converted.

**Review by JVThomas — APPROVED** [review]
Other than pending API doc updates (like a brief explanation of why both ids are used for user) the changes seem good to me.

**Review by quazi-h — APPROVED** [review]
PR looks good to me, makes sense that we should still populate dacUserId for the UI until we update it to point at userId.

**Pull request merged**